corehq/apps/hqwebapp/static/hqwebapp/js/urllib.js

```diff
@@ -0,0 +1,37 @@
+var COMMCAREHQ_URLS = {};
+hqDefine('hqwebapp/js/urllib.js', function () {
+    // http://stackoverflow.com/a/21903119/240553
+    var getUrlParameter = function (sParam) {
+        return getUrlParameterFromString(sParam, window.location.search);
+    };
+    var getUrlParameterFromString = function (sParam, sSearch) {
+        var sPageURL = decodeURIComponent(sSearch.substring(1)),
+            sURLVariables = sPageURL.split('&'),
+            sParameterName,
+            i;
+
+        for (i = 0; i < sURLVariables.length; i++) {
+            sParameterName = sURLVariables[i].split('=');
+
+            if (sParameterName[0] === sParam) {
+                return sParameterName[1] === undefined ? true : sParameterName[1];
```

Comment:
That _is_ interesting. Great question. JS console gives

```
> decodeURIComponent('drink=gin%20%26%20tonics&food=eggplant%20parm')
"drink=gin & tonics&food=eggplant parm"
```

so you're right that's a bug. A bug that would have probably taken years to hit, but a bug none the less, and an easy one to fix.